### DISCUSSION TITLE
http发送get请求收到服务器回复之后几分钟之后才返回

### DISCUSSION THREAD

**Discussion by SheepQi** [cloudwu/skynet#1913] [General]
最近在接入twitch的sdk，需要发送一个get请求到twitch。
在测试过程中发现在返回的header内没有content-length的情况下，会长期的阻塞在internal,lua的recvbody函数中。

o(╥﹏╥)o，也看了注释网址内的说明，好像按规范是得这么搞。但是我拿curl和其他工具测都是立即返回的。有没有大佬懂http这块的可以解惑下，这个是属于BUG呢还是说需要自己改点什么或者设置下什么？

注释内的网址:https://stackoverflow.com/questions/15991173/is-the-content-length-header-required-for-a-http-1-0-response

**Comment by cloudwu** [comment]
没有 content-length 时，相当于内容是无限长的，需要等发送方自己断开 TCP 连接。

> So to answer the question: When no Content-Length is received, the client keeps reading until the server closes the connection.

你要查看为什对方没有发送 TCP 断开。

**Reply by cloudwu** [reply]
如果对方需要等你先发 close ，那么 skynet 目前的 httpc 需要改进，发送完请求就先发一个 close 才行。你需要在合适的时候发送 `socket.shutdown()` https://github.com/cloudwu/skynet/blob/master/lualib/skynet/socket.lua#L270

大致在这里

https://github.com/cloudwu/skynet/blob/master/lualib/http/internal.lua#L195

目前这个封装没有给 接口加 shutdown ，你可以先尝试改，改对了可以提 pr 。

**Comment by SheepQi** [comment]
感谢云大的回复，查了好几天之后发现是个乌龙o(╥﹏╥)o，是发送的代码写的有问题。
